services/core-api/app/api/mines/documents/mine_document_search_util.py

```diff
@@ -64,3 +66,31 @@ def filter_by(cls, mine_guid, project_guid=None, major_mine_application_guid=Non
                 .filter(ProjectDecisionPackage.project_decision_package_guid == project_decision_package_guid)
 
         return qy.all()
+
+    @classmethod
+    def find_by_document_name_and_project_guid(cls, document_name, project_guid=None):
+        """
+        Find Mine Documents by the document_name and the project_guid.
+        """
+        query = db.session.query(MineDocument)
+        
+        if project_guid is not None:
+            # Create aliases for the joined tables
+            md_alias = aliased(MineDocument)
+            psdx_alias = aliased(ProjectSummaryDocumentXref)
+            ps_alias = aliased(ProjectSummary)
+            p_alias = aliased(Project)
+            
+            query = query\
+                .select_from(md_alias)\
+                .with_entities(md_alias.document_name, md_alias.mine_document_guid, p_alias.project_guid, md_alias.is_archived, \
+                    md_alias.mine_guid, md_alias.document_class, md_alias.update_timestamp, md_alias.update_user)\
+                .filter(md_alias.document_name == document_name, md_alias.deleted_ind == False)\
```

Comment:
I think we should probably make this a case insensitive check so that `test.pdf` and `Test.pdf` are considered the same, as that's how most filesystems treat them.

Just a proposal to make this a little bit more readable would be to make use of SQLAlchemy's ability to recognize joining fields

So this could be something like:

```
query = db.session.query(MineDocument)
...
query = query.
    .join(ProjectSummaryDocumentXref)
    .join(ProjectSummary)
    .join(Project)
    .filter(MineDocument.document_name == document_name,...)
    .filter(ProjectSummary.project_guid=project_guid)
```

Reply:
Updated accordingly.